Code/GraphMol/SmilesParse/catch_tests.cpp

```diff
@@ -2520,3 +2520,99 @@ TEST_CASE("ensure unused features are not used") {
     CHECK(smiles == "C[C@H]1C[CH]CC(C[C@@H](C)[C@@H](C)O)C1");
   }
 }
+
+std::unique_ptr<ROMol> getSmartsRootedAtAtom(const ROMol &mol, int root_idx) {
+  bool doIsomericSmarts = true;
+  auto smarts = MolToSmarts(mol, doIsomericSmarts, root_idx);
+  return std::unique_ptr<ROMol>{SmartsToMol(smarts)};
+}
+
+TEST_CASE("Test rootedAtAtom argument", "[smarts]") {
+  SubstructMatchParameters sssparams;
+  sssparams.useChirality = GENERATE(false, true);
+  CAPTURE(sssparams.useChirality);
+
+  SECTION("fully substituted chiral center in linear mol") {
+    auto mol1 = "C[C@](O)(F)CCCl"_smiles;
+    auto mol2 = "C[C@](F)(O)CCCl"_smiles;
+    REQUIRE(mol1);
+    REQUIRE(mol2);
+    REQUIRE(mol1->getNumAtoms() == 7);
+    REQUIRE(mol2->getNumAtoms() == 7);
+
+    auto root_idx = GENERATE(range(-1, 6));
+    CAPTURE(root_idx);
+    auto qmol1 = getSmartsRootedAtAtom(*mol1, root_idx);
+    auto qmol2 = getSmartsRootedAtAtom(*mol2, root_idx);
+
+    CHECK(SubstructMatch(*qmol1, *mol1, sssparams).size() == 1);
```

Comment:
Maybe I've misunderstood the purpose of these tests, but it looks like the calls are backwards,
The query molecule is normally the second argument, so I would expect:
```suggestion
    CHECK(SubstructMatch(*mol1, *qmol1, sssparams).size() == 1);
```

Reply:
Ugh, that's my bad. Seems I don't even remember how to use `SubstructMatch()`. I wonder if that's the reason it was that easy to get it working...

Reply:
Hum, the test still passes after fixing the order of the arguments.